### PR TITLE
CIRC-1725: Bundle overnight automated fee/fine charge notices

### PR DESCRIPTION
**Pull request by OleksandrVidinieiev**
## Purpose
Bundle overnight (not-real-time) automated fee/fine charge notices the same way we do it for overnight due-date notices.

Resolves [CIRC-1725](https://issues.folio.org/browse/CIRC-1725).

## Approach
Most of the code from `GroupedLoanScheduledNoticeHandler` was extracted into an abstract parent `GroupedScheduledNoticeHandler`, which is also extended by a new handler - `GroupedFeeFineScheduledNoticeHandler`.

#### TODOS and Open Questions
- [x] Check logging

### CODE REVIEW

File: src/test/java/api/loans/AgedToLostScheduledNoticesProcessingTests.java [hunk 1]
```diff
@@ -526,6 +530,117 @@ void patronNoticeForAdjustmentOfFullyPaidLostItemFeeIsCreatedAndProcessed() {
...
+
+    verifyBundledFeeFineNotice(UPON_AT_TEMPLATE_ID, userId, Map.of(
+      firstLostItemFeeAccount, firstLoan,
+      firstProcessingFeeAccount, firstLoan,
+      secondLostItemFeeAccount, secondLoan,
+      secondProcessingFeeAccount, secondLoan
+    ));
+  }
+
+  private static JsonObject getAccount(Collection<JsonObject> accounts, UUID loanId, String feeFineType) {
```
Comment (roman-barannyk) [inline src/test/java/api/loans/AgedToLostScheduledNoticesProcessingTests.java:626]:
```suggestion
  private static JsonObject getAccount(Collection<JsonObject> accounts, UUID loanId, 
    String feeFineType) {
```

File: src/test/java/api/loans/AgedToLostScheduledNoticesProcessingTests.java [hunk 2]
```diff
@@ -566,6 +681,44 @@ private void checkSentFeeFineNotices(AgeToLostResult agedToLostResult,
...
+    assertThat(FakeModNotify.getSentPatronNotices(), hasItem(
+      hasEmailNoticeProperties(userId, templateId,
+        allOf(
+          toStringMatcher(getUserContextMatchers(usersClient.get(userId))),
+          JsonPathMatchers.hasJsonPath("feeCharges[*]",  hasItems(
+            getMatchersForBundledFeeFineNotice(accountsToLoans)))
+        ))));
+  }
+
+  private Matcher[] getMatchersForBundledFeeFineNotice(Map<JsonObject, AgeToLostResult> accountsToLoans) {
```
Comment (roman-barannyk) [inline src/test/java/api/loans/AgedToLostScheduledNoticesProcessingTests.java:697]:
```suggestion
  private Matcher[] getMatchersForBundledFeeFineNotice(
    Map<JsonObject, AgeToLostResult> accountsToLoans) {
```

File: src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java [hunk 3]
```diff
@@ -289,6 +288,13 @@ public static JsonObject createFeeFineChargeNoticeContext(Account account, Loan
       .put(FEE_CHARGE, createFeeChargeContext(account, chargeAction));
   }
 
+  public static JsonObject createFeeFineChargeNoticeContextWithoutUser(Account account, Loan loan,
+    FeeFineAction chargeAction) {
+
+    return createLoanNoticeContextWithoutUser(loan)
+      .put(FEE_CHARGE, createFeeChargeContext(account, chargeAction));
+  }
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java:296]: What if fee/fine is not linked to any loan?
What if it's linked to an item but not to a loan?

File: src/test/java/api/loans/AgedToLostScheduledNoticesProcessingTests.java (hunk 1 above)
Comment (OleksandrVidinieiev) [reply src/test/java/api/loans/AgedToLostScheduledNoticesProcessingTests.java:626]: That's one line

File: src/test/java/api/loans/AgedToLostScheduledNoticesProcessingTests.java (hunk 2 above)
Comment (OleksandrVidinieiev) [reply src/test/java/api/loans/AgedToLostScheduledNoticesProcessingTests.java:697]: Also one line :)

**Review by roman-barannyk — APPROVED** [review]

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**